shell/browser/net/preconnect_manager_factory.cc

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2019 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/net/preconnect_manager_factory.h"
+
+#include <memory>
+
+#include "chrome/browser/predictors/preconnect_manager.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/keyed_service/content/browser_context_keyed_service_factory.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/storage_partition.h"
+
+namespace {
+class StubDelegate : public predictors::PreconnectManager::Delegate {
+ public:
+  void PreconnectFinished(
+      std::unique_ptr<predictors::PreconnectStats> stats) override {}
+};
+
+class PreconnectManagerWrapper : public KeyedService {
+ public:
+  PreconnectManagerWrapper(
+      base::WeakPtr<predictors::PreconnectManager::Delegate> delegate,
+      content::BrowserContext* context)
+      : ptr_(new predictors::PreconnectManager(
+            delegate,
+            reinterpret_cast<Profile*>(context))) {
```

Comment:
preconnect manager uses  AsTestingProfile() for tests in PreconnectManager::GetNetworkContext().
This can be removed by small patch.

Reply:
Ah, right, i got confused by the fact that it's in the "content::BrowserContext implementation" section in https://cs.chromium.org/chromium/src/chrome/browser/profiles/profile.h?q=astestingprofile&sq=package:chromium&g=0&l=151. Yeah, I think it would be better to remove that dependency with a patch.

Reply:
added minimal patch